### ISSUE TITLE
Request some details about i-Code V1

### ISSUE THREAD

**Issue by charlesxu90** [microsoft/i-Code#85]
Dear @ziyi-yang and @yuwfan , 
I'm trying to use a similar idea on peptide representation learning. But several details are missing from your context. So I sent this email, hoping to get a response from you.

1. What **representation** did you use in the **single modality task**? The **embedding after the fusion layer or the embedding after the text encoder**? This is not clearly explained in the context, but I think it should influence the results a lot. 
2. I think you should have done some **finetuning using the pretrained i-Code model**. How did you do that? **What classification head did you add?** A linear layer after the embedding or some other dropout tricks as well?
3. In your **calculation of vl, vs, and ls loss**, what **feature** did you use as inputs? The **embedding after the fusion layer**? If so, for the merged attention case, **did you separate each of the text, vision, and speech features from fused embedding?** This seems weird to me and not clear if these features still persist the information it originally contains.

**Comment by charlesxu90** [comment]
In this paper, you mentioned that you're using pre-trained encoders to encode text, image, video and speech, and by adding fusion these embeddings you managed to **obtain better representations than single modal encoders like DeBERTa V3 on single modal task**.

For me, this is interesting, because for most multimodal papers that I have read, they seem to just use good representations from text to supervise images or videos. Your paper shows that other models might help to improve text as well, which indicates that **multimodal learning is not just casting information from a good modality to others, but also jointly constraining the embedding space from multimodal inputs to learn a better embedding.**

**Comment by charlesxu90** [comment]
Besides, I found the initialization of the fusion layers seems to be very important. Otherwise, the model might not converge well. Usually, it would have 'nan' in `loss.backward()`. 

How did you manage to initialize it?

**Comment by ziyi-yang** [comment]
> 

1. After the text encoder.
2. A classification head after the embedding, similar to those in BERT.
3. (1) Yes, The embedding after the fusion layer. (2) No, we didn't separate the features of merge-attention.

**Comment by ziyi-yang** [comment]
> In this paper, you mentioned that you're using pre-trained encoders to encode text, image, video and speech, and by adding fusion these embeddings you managed to **obtain better representations than single modal encoders like DeBERTa V3 on single modal task**.
> 
> For me, this is interesting, because for most multimodal papers that I have read, they seem to just use good representations from text to supervise images or videos. Your paper shows that other models might help to improve text as well, which indicates that **multimodal learning is not just casting information from a good modality to others, but also jointly constraining the embedding space from multimodal inputs to learn a better embedding.**

The text-only performance is obtained with the deberta-v3 encoder that is updated in i-Code pretraining.

**Comment by ziyi-yang** [comment]
> Besides, I found the initialization of the fusion layers seems to be very important. Otherwise, the model might not converge well. Usually, it would have 'nan' in `loss.backward()`.
> 
> How did you manage to initialize it?

In the merge attention case, we use BertEncoder (from transformers.models.bert.modeling_bert import BertEncoder) and its default parameter initialization.

**Comment by charlesxu90** [comment]
Thanks a lot for the reply. Really appreciated.
